wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientRequestStrategy.swift

```diff
@@ -323,10 +323,11 @@ public final class UserClientRequestStrategy: ZMObjectSyncStrategy, ZMObjectStra
                 return false
             } else if clientUpdateStatus?.currentPhase == .waitingForPrekeys {
                 clientUpdateStatus?.willGeneratePrekeys()
+                let nextPrekeyIndex = UInt16(userClient.preKeysRangeMax) + 1
```

Comment:
If anyone is wondering: the `preKeysRangeMax` is set on a successful upload of preKeys.

Reply:
@typfel question is there a risk the index we arrive at this point before the previous generation task below is finished (async) ?

Reply:
> @typfel question is there a risk the index we arrive at this point before the previous generation task below is finished (async) ?

You are asking if it's possible that we have overlapping tasks generating new prekeys? No, because It's controlled to happen serially by the `ClientUpdateStatus.currentPhase`. We only enter this code path if the currentPhase == `.waitingForPrekeys` and while the task is running currentPhase == `.generatingPrekeys` and afterwards currentPhase == `.done` until the prekeys have been uploaded.